listenstore/listenstore/listenstore.py

```diff
@@ -104,21 +110,117 @@ class ListenStore(object):
 
     def __init__(self, conf):
         self.log = logging.getLogger(__name__)
-        self.replication_factor = conf.get("replication_factor", 1)
-        self.keyspace = conf.get("cassandra_keyspace", "listenbrainz")
-        host = conf.get("cassandra_server", "localhost:9092")
+
+    def max_id(self):
+        return int(self.MAX_FUTURE_SECONDS + calendar.timegm(time.gmtime()))
+
+    def fetch_listens(self, uid, from_id=None, to_id=None, limit=None):
+        """ Setup from_id, to_id, and limit for fetching listens
+        """
+        if from_id and to_id:
+            raise ValueError("You cannot specify from_id and to_id at the same time.")
+
+        if from_id:
+            order = ORDER_ASC
+        else:
+            order = ORDER_DESC
+
+        precision = 'month'
+        if from_id is None:
+            from_id = MIN_ID
+        if to_id is None:
+            to_id = self.max_id()
+        return self.fetch_listens_from_database(uid, from_id, to_id, limit, order, precision)
```

Comment:
Also `fetch_listens_from_storage` or something like that would be a more appropriate name, since you don't know where the actual implementation will store listens. It might be in memory or a file instead of database.